auth-jwt-service/src/main/java/io/stargate/auth/jwt/AuthJwtService.java

```diff
@@ -69,6 +79,148 @@ public StoredCredentials validateToken(String token) throws UnauthorizedExceptio
+   *
+   * @param action A {@link QueryBuilder} object to be executed and authorized against a JWT.
+   * @param token The authenticated JWT token to use for authorization.
+   * @param primaryKeyValues A list of primary key values that will be used in the query and should
+   *     be authorized against the JWT.
+   * @param tableMetadata The {@link Table} that will be queried against.
+   * @return On success will return the result of the query and otherwise will return an exception
+   *     relating to the failure to authorize.
+   * @throws Exception An exception relating to the failure to authorize.
+   */
+  @Override
+  public ResultSet executeDataReadWithAuthorization(
+      Callable<ResultSet> action, String token, List<String> primaryKeyValues, Table tableMetadata)
+      throws Exception {
+    JSONObject stargateClaims = extractClaimsFromJWT(token);
+
+    preCheckDataReadWrite(stargateClaims, primaryKeyValues, tableMetadata);
+
+    ResultSet result = action.call();
+
+    if (result == null) {
+      return null;
+    }
+
+    postCheckDataRead(stargateClaims, result);
+
+    return result;
+  }
+
+  /**
+   * TODO
+   *
+   * @param action
+   * @param token
+   * @param primaryKeyValues
+   * @param tableMetadata
+   * @return
+   * @throws Exception
+   */
+  @Override
+  public ResultSet executeDataWriteWithAuthorization(
+      Callable<ResultSet> action, String token, List<String> primaryKeyValues, Table tableMetadata)
+      throws Exception {
+    JSONObject stargateClaims = extractClaimsFromJWT(token);
+
+    preCheckDataReadWrite(stargateClaims, primaryKeyValues, tableMetadata);
+
+    // Just return the result. No value in doing a post check since we can't roll back anyway.
+    return action.call();
+  }
+
+  @Override
+  public ResultSet executeSchemaReadWithAuthorization(
+      Callable<ResultSet> action, String token, String keyspace, String table) throws Exception {
+    JSONObject stargateClaims = extractClaimsFromJWT(token);
+
+    preCheckSchemaRead(stargateClaims, keyspace, table);
+
+    // TODO: [doug] 2020-11-2, Mon, 17:23 Finish implementing
+    return action.call();
+  }
+
+  @Override
+  public ResultSet executeSchemaWriteWithAuthorization(
+      Callable<ResultSet> action, String token, String keyspace, String table) throws Exception {
+    JSONObject stargateClaims = extractClaimsFromJWT(token);
+
+    preCheckSchemaWrite(stargateClaims, keyspace, table);
+
+    // Just return the result. No value in doing a post check since we can't roll back anyway.
+    return action.call();
+  }
+
+  private JSONObject extractClaimsFromJWT(String token) throws JSONException {
+    // Grab the custom claims from the JWT. It's safe to work with the JWT as a plain Base64 encoded
+    // json object here since by this point we've already authenticated the request.
+    String[] parts = token.split("\\.");
+    String decodedPayload = new String(Base64.getUrlDecoder().decode(parts[1]));
```

Comment:
I would add a simple check (parts.size() == 2) and if it is not met, throw `IllegalArgumentException`

Reply:
To be extra safe I can. As part of authentication it'll make sure that the JWT is properly formatted.